test/popup.spec.js

```diff
@@ -284,52 +276,29 @@ module.exports.describe = function({testRunner, expect, playwright, CHROMIUM, WE
         const win = window.open('about:blank');
         win.alert('hello');
       });
-      const popupEvent = await page.waitForEvent('popup');
-      const popup = await popupEvent.page();
+      const popup = await page.waitForEvent('popup');
       const dialog = await popup.waitForEvent('dialog');
       expect(dialog.message()).toBe('hello');
       await dialog.dismiss();
       await evaluatePromise;
       await context.close();
     });
-    it.fail(FFOX)('should work with empty url', async({browser}) => {
+    it('should work with empty url', async({browser}) => {
       const context = await browser.newContext();
       const page = await context.newPage();
       const [popup] = await Promise.all([
-        page.waitForEvent('popup').then(e => e.page()),
+        page.waitForEvent('popup'),
         page.evaluate(() => window.__popup = window.open('')),
       ]);
       expect(await page.evaluate(() => !!window.opener)).toBe(false);
       expect(await popup.evaluate(() => !!window.opener)).toBe(true);
       await context.close();
     });
-    it('should work with empty url and nowait', async({browser}) => {
-      const context = await browser.newContext();
-      const page = await context.newPage();
-      const [popup] = await Promise.all([
-        page.waitForEvent('popup').then(e => e.page({ waitUntil: 'nowait' })),
-        page.evaluate(() => window.__popup = window.open()),
-      ]);
-      expect(await page.evaluate(() => !!window.opener)).toBe(false);
-      expect(await popup.evaluate(() => !!window.opener)).toBe(true);
-      await context.close();
-    });
-    it('should work with noopener', async({browser}) => {
-      const context = await browser.newContext();
-      const page = await context.newPage();
-      const [popup] = await Promise.all([
-        page.waitForEvent('popup').then(e => e.page()),
-        page.evaluate(() => window.__popup = window.open('about:blank', null, 'noopener')),
-      ]);
-      expect(await page.evaluate(() => !!window.opener)).toBe(false);
-      expect(await popup.evaluate(() => !!window.opener)).toBe(false);
-      await context.close();
-    });
-    it('should work with noopener and nowait', async({browser}) => {
+    it('should work with noopener and about:blank', async({browser}) => {
```

Comment:
Same here - we had two tests to check that we can resolve early, and also can wait for load. Let's have two?